host/tests/dsCompositeIn_L3_Tests/dsCompositeIn_test4_VerifySignal_Callback.py

```diff
@@ -0,0 +1,197 @@
+        if url is not None:
+            self.downloadToDevice(url, self.deviceDownloadPath, self.rackDevice)
+            for streampath in url:
+                self.testStreams.append(os.path.join(self.deviceDownloadPath, os.path.basename(streampath)))
+
+    def testCleanAssets(self):
+        """
+        Removes the assets copied to the dut.
+
+        Args:
+            None.
+        """
+        self.deleteFromDevice(self.testStreams)
+
+    def testRunPrerequisites(self):
+        """
+        Executes prerequisite commands listed in the test setup configuration file on the DUT.
+
+        Args:
+            None
+        """
+
+        # Run commands as part of test prerequisites
+        cmds = self.testSetup.assets.device.test4_VerifySignal_Callback.execute
+        if cmds is not None:
+            for cmd in cmds:
+                self.writeCommands(cmd)
+
+     #TODO: Current version supports only manual verification.
+    def CheckDeviceStatus(self, manual=False, port_type:str=0):
+        """
+        Verifies whether Composite Source Device connected or not
+
+        Args:
+            port_type (str) : CompositeIn port
+            manual (bool, optional): Manual verification (True: manual, False: other verification methods).
+                                     Defaults to other verification methods
+
+        Returns:
+            bool
+        """
+        if manual == True:
+            return self.testUserResponse.getUserYN(f"Check if CompositeIn source is connected to {port_type} and press Y:")
+        else :
+            #TODO: Add automation verification methods
+            return False
+
+    def testFunction(self):
+        """
+        The main test function that verifies signal status of CompositeIn device.
+
+        This function:
+        - Downloads necessary assets.
+        - Runs prerequisite commands.
+        - Verifies CompositeIn signal status through callbacks.
+
+        Returns:
+            bool: Final result of the test.
+        """
+
+        # Download the assets listed in test setup configuration file
+        self.testDownloadAssets()
+
+        # Run Prerequisites listed in the test setup configuration file
+        self.testRunPrerequisites()
+
+        # Create the dsCompositeIn class
+        self.testdsCompositeIn = dsCompositeInClass(self.deviceProfile, self.hal_session)
+
+        self.log.testStart("test4_VerifySignal_Callback", '1')
+
+        # Initialize the dsCompositeIn module
+        self.testdsCompositeIn.initialise()
+
+        # Loop through the supported CompositeIn ports
+        for port in self.testdsCompositeIn.getSupportedPorts():
+            self.log.stepStart(f'Select {port} Port')
+
+            # Check the CompositeIn device connected to is active
+            portstr = f"dsCOMPOSITE_IN_PORT_{port}"
+            result = self.CheckDeviceStatus(True,portstr)
+            self.log.stepResult(result,f'CompositeIn Device connected {result} on {portstr}')
```

Comment:
use self.log.step instead of stepResult .

Reply:
Step and step result of that... used similar like audio